### PR TITLE
Test: Types - ConsensusData

### PR DESCRIPTION
**Pull request by MatheusFranco99**
Implement tests for `ConsensusData`.

Tests:
- [x] Attestation consensus data encoding
- [x] Aggregation consensus data encoding
- [x] Proposer consensus data encoding
- [x] Blinded proposer consensus data encoding
- [x] Sync committee consensus data encoding
- [x] Sync committee contribution consensus data encoding
---
- [x] Invalid duty
---
- [x] Valid proposer consensus data with Capella block
- [x] Valid proposer consensus data with blinded Capella block
- [x] Proposer consensus data with no justification
- [x] Invalid proposer consensus data with for Capella block
- [x] Invalid proposer consensus data with for blinded Capella block
---
- [x] Valid attestation consensus data
- [x] Invalid attestation consensus data with pre-consensus justification
- [x] Invalid attestation consensus data with wrong data
---
- [x] Valid aggregation consensus data
- [x] Invalid aggregation consensus data with no pre-consensus justification
- [x] Invalid aggregation consensus data with wrong data
---
- [x] Valid sync committee consensus data
- [x] Invalid sync committee consensus data with pre-consensus justification
- [x] Invalid sync committee consensus data with wrong data
---
- [x] Valid sync committee contribution consensus data
- [x] Invalid sync committee contribution  consensus data with no pre-consensus justification
- [x] Invalid sync committee contribution consensus data with wrong data

Note: some files were deleted because they had duplicated test specifications.

### CODE REVIEW

**Review by GalRogozinski — COMMENTED** [review]
I think I review this after we conclude on value check

File: types/spectest/tests/consensusdata/capella_blinded_block_validation.go [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
 
+import (
+	"github.com/attestantio/go-eth2-client/spec"
+	"github.com/bloxapp/ssv-spec/types/testingutils"
+)
+
 // CapellaBlindedBlockValidation tests a valid consensus data with capella blinded block
-func CapellaBlindedBlockValidation() *SpecTest {
-	panic("implement")
+func CapellaBlindedBlockValidation() *ConsensusDataTest {
```
Comment (GalRogozinski) [inline types/spectest/tests/consensusdata/capella_blinded_block_validation.go:9]: In Deneb PR we should maybe add tests if they are not there?

File: types/spectest/tests/consensusdata/contributions_encoding.go [hunk 2]
```diff
@@ -1,6 +0,0 @@
-package consensusdata
-
-// ContributionsEncoding tests encoding and decoding contributions
-func ContributionsEncoding() *SpecTest {
-	panic("implement")
```
Comment (GalRogozinski) [inline types/spectest/tests/consensusdata/contributions_encoding.go:5]: why is this deleted?

File: types/spectest/tests/consensusdata/invalid_sync_committee_block_validation.go [hunk 3]
```diff
@@ -1,6 +1,23 @@
...
+	cd := types.ConsensusData{
+		Duty:    testingutils.TestingSyncCommitteeDuty,
+		Version: spec.DataVersionCapella,
+		DataSSZ: []byte{1},
+	}
+
+	return &ConsensusDataTest{
+		Name:          "invalid sync committee",
+		ConsensusData: cd,
+		ExpectedError: "could not unmarshal ssz: expected buffer of length 32 receiced 1",
```
Comment (GalRogozinski) [inline types/spectest/tests/consensusdata/invalid_sync_committee_block_validation.go:21]: there is a typo in the error.. it is from our code?

File: types/spectest/tests/consensusdata/invalid_sync_committee_cntribution_validation.go [hunk 4]
```diff
@@ -1,6 +1,16 @@
...
-	panic("implement")
+func InvalidSyncCommitteeContributionValidation() *ConsensusDataTest {
+
+	cd := testingutils.TestSyncCommitteeContributionConsensusData
+	cd.DataSSZ = testingutils.TestingAttestationDataBytes
+
+	return &ConsensusDataTest{
+		Name:          "invalid sync committee contribution",
+		ConsensusData: *cd,
+		ExpectedError: "could not unmarshal ssz: four",
```
Comment (GalRogozinski) [inline types/spectest/tests/consensusdata/invalid_sync_committee_cntribution_validation.go:14]: you know what "four" means here?
it comes from ssz lib

File: types/spectest/tests/consensusdata/proposer_justifications.go [hunk 5]
```diff
@@ -1,6 +0,0 @@
-package consensusdata
-
-// ProposerJustifications tests a valid consensus data with proposer justifications
-func ProposerJustifications() *SpecTest {
-	panic("implement")
```
Comment (GalRogozinski) [inline types/spectest/tests/consensusdata/proposer_justifications.go:5]: why is this deleted?

**Review by GalRogozinski — COMMENTED** [review]
LGTM just answer my questions and I'll approve

File: types/spectest/tests/consensusdata/invalid_sync_committee_cntribution_validation.go (hunk 4 above)
Comment (y0sher) [reply types/spectest/tests/consensusdata/invalid_sync_committee_cntribution_validation.go:14]: seems to be like an error code or something, that's pretty funny, from the fastssz lib:
```go
		if offset > endOffset {
			return fmt.Errorf("four")
		}
		if endOffset > size {
			return fmt.Errorf("five")
		}
```

File: types/spectest/tests/consensusdata/sync_committee_contrib_no_justifications.go [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import (
+	"github.com/attestantio/go-eth2-client/spec"
+	"github.com/bloxapp/ssv-spec/types"
+	"github.com/bloxapp/ssv-spec/types/testingutils"
+)
+
+// SyncCommitteeContributionNoJustifications tests an invalid consensus data with no sync committee contribution pre-consensus justifications
+func SyncCommitteeContributionNoJustifications() *ConsensusDataTest {
+
+	// To-do: add error when pre-consensus justification check is added.
```
Comment (y0sher) [inline types/spectest/tests/consensusdata/sync_committee_contrib_no_justifications.go:12]: is this a test todo or we missing this check in general?

**Review by y0sher — APPROVED** [review]

File: types/spectest/tests/consensusdata/sync_committee_contrib_no_justifications.go (hunk 6 above)
Comment (GalRogozinski) [reply types/spectest/tests/consensusdata/sync_committee_contrib_no_justifications.go:12]: when we bring back pre-consensus justification it shouldn't be empty

**Review by GalRogozinski — APPROVED** [review]
@MatheusFranco99 
I approve but still answer my questions please 🙏

File: types/spectest/tests/consensusdata/capella_blinded_block_validation.go (hunk 1 above)
Comment (MatheusFranco99) [reply types/spectest/tests/consensusdata/capella_blinded_block_validation.go:9]: Yes, we will

File: types/spectest/tests/consensusdata/contributions_encoding.go (hunk 2 above)
Comment (MatheusFranco99) [reply types/spectest/tests/consensusdata/contributions_encoding.go:5]: It's already inside `encoding.go` (duplicate)

File: types/spectest/tests/consensusdata/invalid_sync_committee_block_validation.go (hunk 3 above)
Comment (MatheusFranco99) [reply types/spectest/tests/consensusdata/invalid_sync_committee_block_validation.go:21]: Not from our code

File: types/spectest/tests/consensusdata/proposer_justifications.go (hunk 5 above)
Comment (MatheusFranco99) [reply types/spectest/tests/consensusdata/proposer_justifications.go:5]: Duplicate (same as proposer_valid test)
